src/vnsw/opencontrail-vrouter-netns/opencontrail_vrouter_netns/vrouter_hyperv.py

```diff
@@ -0,0 +1,469 @@
+        if resp.status_code != requests.codes.ok:
+            error_str = resp.text
+            try:
+                err = json.loads(resp.text)
+                error_str = err['error']
+            except Exception:
+                pass
+            raise ValueError(error_str)
+
+
+    def _add_port_to_agent(self, nic, display_name=None):
+        if self.PORT_TYPE == "NovaVMPort":
+            port_type_value = 0
+        elif self.PORT_TYPE == "NameSpacePort":
+            port_type_value = 1
+        payload = {"ip-address": str(nic['ip'].ip), "tx-vlan-id": -1,
+                   "display-name": display_name, "id": nic['uuid'],
+                   "instance-id": self.vm_uuid, "ip6-address": '',
+                   "rx-vlan-id": -1,
+                   "system-name": self._get_wingw_iface_name(nic['uuid']),
+                   "vn-id": '', "vm-project-id": '',
+                   "type": port_type_value, "mac-address": str(nic['mac'])}
+        json_dump = json.dumps(payload)
+        # TODO no agent - uncomment when it works
+        #self._request_to_agent(self.BASE_URL, 'post', json_dump) 
+
+
+    def _delete_port_from_agent(self, nic):
+        url = self.BASE_URL + "/" + nic['uuid']
+        # TODO no agent - uncomment when it works
+        # self._request_to_agent(url, 'delete', None)
+
+
+    def _get_wingw_iface_name(self, uuid_str):
+        pass
+        # TODO no agent - uncomment and fix when it works
+        #return (self.TAP_PREFIX + uuid_str)[:self.DEV_NAME_LEN]
+
+
+    def _exec_ssh_command(self, command):
+        _, stdout, _ = self.ssh_client.exec_command(command)
+        return stdout.channel.recv_exit_status()
+
+
+    def _setup_snat(self):
+        self._exec_ssh_command("sudo sysctl -w net.ipv4.ip_forward=1")
+        self._exec_ssh_command("sudo ip link set dev {} up"
+                               .format(self.nic_left['name']))
+        self._exec_ssh_command("sudo ip link set dev {} up"
+                               .format(self.nic_right['name']))
+        self._exec_ssh_command("sudo iptables -t nat -F")
+        self._exec_ssh_command("sudo iptables -t nat -A POSTROUTING -o {} "
+                               "-j MASQUERADE"
+                               .format(self.nic_right['name']))
+        self._exec_ssh_command("sudo iptables -A FORWARD -i {} -o {} -m state "
+                               "--state RELATED,ESTABLISHED -j ACCEPT"
+                               .format(self.nic_right['name'],
+                                       self.nic_left['name']))
+        self._exec_ssh_command("sudo iptables -A FORWARD -i {} -o {} -j ACCEPT"
+                               .format(self.nic_left['name'],
+                                       self.nic_right['name']))
+        self._exec_ssh_command("sudo /etc/init.d/iptables-persistent save")
+
+
+class VRouterHyperV(object):
+    """Create or destroy a Hyper-V Gateway VM for NAT between two virtual 
+    networks.
+    """
```

Comment:
I think you should decide whether you put `"""` in the same line or in a separate like. Like that for example:
```
"""
Create or destroy a Hyper-V Gateway VM for NAT between two virtual 
networks.
"""
```